pkg/reconciler/instances/connectivityproxy/action.go

```diff
@@ -25,6 +25,11 @@ func (a *CustomAction) Run(context *service.ActionContext) error {
 	if context.Task.Type == model.OperationTypeDelete {
 		context.Logger.Info("Requested cluster removal - removing component")
 		if err := a.Commands.Remove(context); err != nil {
+			context.Logger.Error("Failed to remove Connectivity Proxy: %v", err)
+			return err
+		}
+		if err := a.Commands.RemoveIstioSecrets(context); err != nil {
```

Comment:
This does not need to be called since `RemoveIstioSecrets` is called by Commands.Remove method (see L139 in [pkg/reconciler/instances/connectivityproxy/command.go)  After this  change you can delete `RemoveIstioSecrets`   method from the interface and itd mocked conterpart. I would suggest  make this method as package private